cli/cmd/inject.go

```diff
@@ -479,3 +490,12 @@ func parsePortRanges(portRanges []*cfg.PortRange) string {
 
 	return strings.TrimSuffix(str, ",")
 }
+
+func getFlagValue(flags []*cfg.Install_Flag, name string) string {
```

Comment:
The reason I included this was so that if the `install` was done with a custom registry, an override at `inject`-time would correctly override that custom registry rather than expecting the default `gcr.io/linkerd-io` for the override.  I believe this should be applied to your prior PR for the proxy and proxy-init as well.  I'll have to look at it again more closely this evening, but I believe that to be the case if I'm not mistaken.

Reply:
@adleong I've confirmed that we do need this and was able to verify with additional unit tests.  Here's the process I'm guarding against:

Install using `linkerd install --registry=install.registry` results in ConfigMap settings like:
```
data:
  proxy: {"imageName":"install.registry/proxy",...}
  install: {"flags":[{"name":"registry", "value":"install.registry"}],...}
```

Inject using `linkerd inject --registry=inject.registry` will still only inject `install.registry/proxy`.

The injected image is not overridden because the current `registryOverride()` method in `install.go` only replaces the _default_ registry, i.e. `gcr.io/linkerd-io`.  In our case, we want to have the new registry override `install.registry` which we only know by looking at the installation flags.

The `registryOverride()` method in `install.go` is fine as it is, but should not be used by `inject.go`.

Reply:
Ah, nice catch then.  It seems to me that this registry replacement logic has always been quite brittle.  Could we simplify things by just using string splitting or a regex to replace the registry portion of the string, without needing to know the previous value?

Reply:
We could for the debug image since it has no other way to override the image name other than the registry portion (i.e. we don't allow a CLI option of `--debug-image`) and therefore we'd replace anything prior to "/debug."  As for proxy and proxy-init, we don't really have that luxury as during install, someone could use `my.custom.registry/myproxy`.  I _think_ this is probably the closest we can get offhand, though still not 100% infallible due to the flexibility allowed with the CLI.

Reply:
Hmmm... there must be something I'm missing here, because I still don't see why we need to know the previous registry value.  This is the flow as I understand it:

Suppose Linkerd is installed with `--registry my-install-registry.io/linkerd-io` and `--proxy-image gcr.io/my-install-image`.  I would expect that the proxy-image flag would take priority and the resulting proxy image name of `gcr.io/my-install-image` would be set into the Linkerd configmap.  Since there is no flag for setting the debug image name, `my-install-registry.io/linkerd-io/debug` is set as the debug image name in the configmap.

Then suppose that inject is run with `--registry my-inject-registry.io/linkerd-io` and `--proxy-image gcr.io/my-inject-image`.  For the proxy, again the proxy image flag takes priority and we use `gcr.io/my-inject-image` as the proxy image name instead of the value in the configmap.  For the debug container, we take the value from the configmap (`my-install-registry.io/linkerd-io/debug`) and replace the registry part (everything up to the last slash) with the registry override from the inject flag: `my-inject-registry.io/linkerd-io/debug`.

Am I missing something about the way this works?

Reply:
@adleong Ok, now I see!  Sorry...wasn't grokking that the last slash truly indicates the end of the registry name; I guess I was thinking of a more convoluted registry and image name where a slash "/" could be part of the image name which cannot be.  

I'll make that change...definitely simplifies the replacement. 👍